libraries/cyclestreets-core/src/main/java/net/cyclestreets/CycleStreetsPreferences.java

```diff
@@ -135,24 +138,44 @@ public static int replanDistance() {
   }
 
   public static boolean uploadSmallImages() {
-    final String resize = uploadSize();
-    if ("640px".equals(resize))
+    if ("640px".equals(uploadSize()))
       return true;
-    if ("big".equals(resize))
+    if ("big".equals(uploadSize()))
       return false;
 
+    return !onFastConnection();
+  }
```

Comment:
@jezhiggins It would be a bit challenging to test all this stuff.

So I'd appreciate a careful second pair of eyes sanity checking through code read, please!